ngraph/frontend/onnx/frontend/src/utils/onnx_internal.cpp

```diff
@@ -81,16 +81,19 @@ void convert_decoded_function(std::shared_ptr<Function> function) {
 }
 
 std::shared_ptr<Function> import_onnx_model(std::shared_ptr<ONNX_NAMESPACE::ModelProto> model_proto,
-                                            const std::string& model_path) {
+                                            const std::string& model_path,
+                                            const std::shared_ptr<ov::frontend::TelemetryExtension>& telemetry) {
     apply_transformations(*model_proto, model_path);
-    Graph graph{model_proto};
+    Graph graph{model_proto, telemetry};
```

Comment:
Regarding the suggestion above - this is where you could send the telemetry data without injecting the telemetry extension to the `Graph` object.